guides/advanced_guides/settings.md

```diff
@@ -0,0 +1,251 @@
+
+## Ranking rules
+
+Built-in ranking rules that **ensure relevancy in search results**. Ranking rules are applied in a default order which can be changed in the settings. You can add or remove rules and change their order of importance.
+
+`rankingRules=[<String>, <String>, ...]`
+
+- `[<String>, <String>, ...]` (Array of strings, see default value below)
+
+  An array of strings that contains the ranking rules sorted by order of importance (arranged from the most important rule to the least important rule).
+
+Default value:
+
+```json
+["typo", "words", "proximity", "attribute", "wordsPosition", "exactness"]
+```
+
+By default, rules are executed in the following order:
+
+#### 1. Typo
+
+Results are sorted by **increasing number of typos**: find documents that match query terms with fewer typos first.
+
+#### 2. Words
+
+Results are sorted by **decreasing number of matched query terms** in each matching document: find documents that contain more occurrences of the query terms first.
+
+#### 3. Proximity
+
+Results are sorted by **increasing distance between matched query terms**: find documents that contain more query terms found close together (close proximity between two query terms) and appearing in the original order specified in the query string first.
+
+#### 4. Attribute
+
+Results are sorted according to **[the order of importance of the attributes](/guides/main_concepts/relevancy.md#attributes-importance)**: find documents that contain query terms in more important attributes first.
+
+#### 5. Words Position
+
+Results are sorted by **the position of the query words in the attributes**: find documents that contain query terms earlier in their attributes first.
+
+#### 6. Exactness
+
+Results are sorted by **the similarity of the matched words with the query words**: find documents that contain exactly the same terms as the ones queried first.
+
+[Learn more about ranking rules](/guides/main_concepts/relevancy.md)
+
+#### Example
+
+Add settings:
+
+```bash
+$ curl \
+  -X POST 'http://localhost:7700/indexes/movies/settings' \
+  --data '{
+      "rankingRules": [
+          "typo",
+          "words",
+          "proximity",
+          "attribute",
+          "wordsPosition",
+          "exactness",
+          "asc(release_date)",
+          "desc(rank)"
+      ]
+  }'
+```
+
+## Distinct attribute
+
+The **value of a field** whose attribute is set as a distinct attribute will always be **unique** in the returned documents.
+
+`distinctAttribute=<String>`
+
+- `<String>` (String, defaults to `null`)
+
+  The field name.
+
+[Learn more about the distinct attribute](/guides/advanced_guides/distinct.md)
+
+#### Example
+
+Add settings:
+
+```bash
+$ curl \
+  -X POST 'http://localhost:7700/indexes/movies/settings' \
+  --data '{
+      "distinctAttribute": "movie_id"
```

Comment:
This is my fault, but yes this example is not clear. We should find a more convincing example.
--- IDEA 1 ----
Given a movie dataset where a document represent a film in a certain language. For example 

```
id: "123456"
title: "Star wars: the return of the jedi"
description: "..."
movie_id: "abbc123"
```

```
id: "7891011"
title: "Star wars: le retour des jedis"
description: "..."
movie_id: "abbc123"
```

Different id's but same movie_id. This will prevent both film to be given back when searching for "Star wars".

---- IDEA 2 ----
For example, instead of using /movies/ we could use /products/ and then use `article_uid`. And explain underneath the example that we have multiple time the same article with different colors so in order to ignore the different colors we search by `article uid`. In which case a search on any article will give you different articles and not 10 times the same with a different color. 

A product would look like this 
```
id: document_id
description: `Leather jacket`
brand: `Lee jeans`
color: `brown`
article_id: `123456`
```
```
id: document_id
description: `Leather jacket`
brand: `Lee jeans`
color: `black`
article_id: `7891011`
```
If we make a distinct on article_id and a search on `Lee leather jacket` we will only receive on of the two document in return. And then, if the user wants more colors he should click on it but that's not our business.

Reply:
I like the jacket example and could use it for the distinct attribute advanced guide. Thx a lot!